### PR TITLE
Bump Cruise Control to latest

### PR DESCRIPTION
**Pull request by Boojapho**
### Type of change

- Enhancement / new feature

### Description

Bump Cruise Control to latest to inherit bug and security fixes

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md
- [x] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
The code looks good. I will run the system tests. Could you please fix the DCO sign-off? The instructions should be under the _Details_ link next to the DCO status.

**Review by samuel-hawker — APPROVED** [review]
LGTM
Quick question, I see that our CC versions in 2.7.x vs 2.8.x were different
image
Was there a reason for this and should this be taken into account for the current PR.
Given this is a minor update, I don't think it matters, but thought it worth mentioning.

**Comment by samuel-hawker** [comment]
^^ The above is more of a question for @scholzj and the other maintainers.

**Comment by tomncooper** [comment]
Thanks for the PR!

This new version adds a new replica movement strategy options from [CC #1559](https://github.com/linkedin/cruise-control/pull/1559) which should be added to the relevant sections of the [documentation](https://strimzi.io/docs/operators/latest/using.html#con-rebalance-str). It will be important to note that:
```
Current PR assumes that if users intend to use the new PrioritizeMinIsrWithOfflineReplicasStrategy
strategy, then minISR-based concurrency adjustment is enabled. Otherwise, the 
PrioritizeMinIsrWithOfflineReplicasStrategy strategy is just a no-op.
```

**Comment by tomncooper** [comment]
> ^^ The above is more of a question for @scholzj and the other maintainers.

As far as I understand it, there is no reason for them to be different. That was just an oversight on our part.

**Comment by Boojapho** [comment]
> The code looks good. I will run the system tests. Could you please fix the DCO sign-off? The instructions should be under the _Details_ link next to the DCO status.

I've fixed the DCO now.

**Comment by Boojapho** [comment]
> Thanks for the PR!
> 
> This new version adds a new replica movement strategy options from [CC #1559](https://github.com/linkedin/cruise-control/pull/1559) which should be added to the relevant sections of the [documentation](https://strimzi.io/docs/operators/latest/using.html#con-rebalance-str). It will be important to note that:
> 
> ```
> Current PR assumes that if users intend to use the new PrioritizeMinIsrWithOfflineReplicasStrategy
> strategy, then minISR-based concurrency adjustment is enabled. Otherwise, the 
> PrioritizeMinIsrWithOfflineReplicasStrategy strategy is just a no-op.
> ```

@tomncooper I think I have updated the docs for what is needed.  Appreciate if you could take a look.

**Review by scholzj — APPROVED** [review]

File: documentation/modules/cruise-control/con-rebalance-performance.adoc [hunk 1]
```diff
@@ -33,11 +33,12 @@ Cluster rebalance performance is also influenced by the _replica movement strate
 By default, Cruise Control uses the `BaseReplicaMovementStrategy`, which simply applies the commands in the order they were generated.
 However, if there are some very large partition reassignments early in the proposal, this strategy can slow down the application of the other reassignments.
 
-Cruise Control provides three alternative replica movement strategies that can be applied to optimization proposals:
+Cruise Control provides four alternative replica movement strategies that can be applied to optimization proposals:
 
 * `PrioritizeSmallReplicaMovementStrategy`: Order reassignments in order of ascending size.
 * `PrioritizeLargeReplicaMovementStrategy`: Order reassignments in order of descending size. 
 * `PostponeUrpReplicaMovementStrategy`: Prioritize reassignments for replicas of partitions which have no out-of-sync replicas.
+* `PrioritizeMinIsrWithOfflineReplicasStrategy`: Prioritize reassignments with (At/Under)MinISR partitions with offline replicas.  This strategy will only work if Cruise Control has been configured with `min_isr_based_concurrency_adjustment` set to true.
```
Comment (tomncooper) [inline documentation/modules/cruise-control/con-rebalance-performance.adoc:41]: We don't support the CC REST API `/admin` endpoint so a Strimzi user cannot set `min_isr_based_concurrency_adjustment`. They would need to set the `concurrency.adjuster.min.isr.check.enabled` in the Cruise Control section of the Kafka CR.

Comment (Boojapho) [reply documentation/modules/cruise-control/con-rebalance-performance.adoc]: Updated to "if `cruiseControl.config.concurrency.adjuster.min.isr.check.enabled` is set to `true` in the `Kafka` custom resource's spec."

**Comment by scholzj** [comment]
@tomncooper Does it look good now? I can deal with the conflict later.

**Review by tomncooper — APPROVED** [review]
LGTM

**Comment by scholzj** [comment]
Thanks for the PR.

**Pull request merged**